### PR TITLE
Fix Windows tests

### PR DESCRIPTION
**Pull request by klauspost**
## Motivation and Context

Replaces #9299

Skips ServerTest for now.

## How to test this PR?

CI.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: cmd/object-api-input-checks.go [hunk 1]
```diff
@@ -50,6 +51,10 @@ func checkBucketAndObjectNames(ctx context.Context, bucket, object string) error
 		logger.LogIf(ctx, ObjectNameInvalid{Bucket: bucket, Object: object})
 		return ObjectNameInvalid{Bucket: bucket, Object: object}
 	}
+	if strings.Contains(object, "\\") {
```
Comment (harshavardhana) [inline cmd/object-api-input-checks.go]: Let's just do this only for Windows @klauspost

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :x: [more...](#43d74e4c-323d-40e1-a564-b99227a15072) |
| mint-gateway-azure.sh | :x: [more...](#3dcdcc8a-6e49-4d51-a7ee-93e1f7a3a9ca) |
<a name="43d74e4c-323d-40e1-a564-b99227a15072"></a>
### 9594-af746de/mint-gateway-s3.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev4.minio.io:31200
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 794f1555456c:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 41 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... FAILED in 20 seconds
{
  "name": "awscli",
  "duration": 4385,
  "function": "aws --endpoint-url http://minio-dev4.minio.io:31200 s3api get-object --bucket awscli-mint-test-bucket-31372 --key awscli-mint-test-bucket-31372-object /tmp/datafile-0-b\n",
  "status": "FAIL",
  "error": "Etag mismatch for multipart 0 byte object"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 18 seconds
(7/15) Running minio-dotnet tests ... done in 28 seconds
(8/15) Running minio-go tests ... done in 29 seconds
(9/15) Running minio-java tests ... done in 26 seconds
(10/15) Running minio-js tests ... done in 35 seconds
(11/15) Running minio-py tests ... done in 1 minutes and 1 seconds
(12/15) Running s3cmd tests ... done in 1 minutes and 4 seconds
(13/15) Running s3select tests ... done in 1 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="3dcdcc8a-6e49-4d51-a7ee-93e1f7a3a9ca"></a>
### 9594-af746de/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:30753
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp bb6a6266b0ab:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 5 minutes and 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 15 seconds
(5/15) Running awscli tests ... done in 2 minutes and 55 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 3 minutes and 6 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 22 seconds
(9/15) Running minio-go tests ... done in 3 minutes and 48 seconds
(10/15) Running minio-java tests ... FAILED in 27 minutes and 3 seconds
{
  "name": "minio-java",
  "function": "listIncompleteUploads(String bucketName, String prefix)",
  "duration": 1471,
  "status": "FAIL",
  "error": "java.io.EOFException >>> [okio.RealBufferedSink.write(RealBufferedSink.java:115), io.minio.HttpRequestBody.writeTo(HttpRequestBody.java:78), okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:94), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:125), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121), okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:264), okhttp3.RealCall.execute(RealCall.java:93), io.minio.MinioClient.execute(MinioClient.java:1097), io.minio.MinioClient.execute(MinioClient.java:1236), io.minio.MinioClient.executePut(MinioClient.java:1410), io.minio.MinioClient.executePut(MinioClient.java:1424), io.minio.MinioClient.uploadPart(MinioClient.java:5492), io.minio.MinioClient.putObject(MinioClient.java:3740), io.minio.MinioClient.putObject(MinioClient.java:3870), FunctionalTest.listIncompleteUploads_test2(FunctionalTest.java:1790), FunctionalTest.runTests(FunctionalTest.java:4052), FunctionalTest.main(FunctionalTest.java:4192)]"
}
(10/15) Running minio-js tests ... done in 2 minutes and 36 seconds
(11/15) Running minio-py tests ... done in 10 minutes and 5 seconds
(12/15) Running s3cmd tests ... FAILED in 59 seconds
{
  "name": "s3cmd",
  "duration": "2330",
  "function": "test_copy_object",
  "status": "FAIL",
  "error": ""
}
(12/15) Running s3select tests ... done in 22 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :x: [more...](#43023e58-8ee7-482a-9b47-80d137869ebd) |
| mint-gateway-azure.sh | :x: [more...](#36c4079e-7c95-4ed4-aff4-a5eda940fb7d) |
<a name="43023e58-8ee7-482a-9b47-80d137869ebd"></a>
### 9594-19bd596/mint-gateway-s3.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev4.minio.io:32259
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 14b1f9fb9b54:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 41 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... FAILED in 20 seconds
{
  "name": "awscli",
  "duration": 4213,
  "function": "aws --endpoint-url http://minio-dev4.minio.io:32259 s3api get-object --bucket awscli-mint-test-bucket-28135 --key awscli-mint-test-bucket-28135-object /tmp/datafile-0-b\n",
  "status": "FAIL",
  "error": "Etag mismatch for multipart 0 byte object"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 17 seconds
(7/15) Running minio-dotnet tests ... done in 29 seconds
(8/15) Running minio-go tests ... done in 24 seconds
(9/15) Running minio-java tests ... done in 25 seconds
(10/15) Running minio-js tests ... done in 30 seconds
(11/15) Running minio-py tests ... done in 1 minutes and 4 seconds
(12/15) Running s3cmd tests ... done in 1 minutes and 8 seconds
(13/15) Running s3select tests ... done in 1 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="36c4079e-7c95-4ed4-aff4-a5eda940fb7d"></a>
### 9594-19bd596/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev4.minio.io:30113
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 6b0396734d4a:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 2 minutes and 32 seconds
(4/15) Running aws-sdk-ruby tests ... done in 15 seconds
(5/15) Running awscli tests ... done in 2 minutes and 9 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 3 minutes and 15 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 20 seconds
(9/15) Running minio-go tests ... done in 3 minutes and 51 seconds
(10/15) Running minio-java tests ... done in 26 minutes and 46 seconds
(11/15) Running minio-js tests ... done in 2 minutes and 41 seconds
(12/15) Running minio-py tests ... done in 10 minutes and 8 seconds
(13/15) Running s3cmd tests ... FAILED in 43 seconds
{
  "name": "s3cmd",
  "duration": "1719",
  "function": "test_copy_object",
  "status": "FAIL",
  "error": ""
}
(13/15) Running s3select tests ... done in 22 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :x: [more...](#c0237056-7b83-491d-ae6a-1d50a4c4310d) |
<a name="c0237056-7b83-491d-ae6a-1d50a4c4310d"></a>
### 9594-726f7d9/mint-gateway-s3.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev3.minio.io:30948
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp fadcae903e98:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 41 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... FAILED in 20 seconds
{
  "name": "awscli",
  "duration": 4453,
  "function": "aws --endpoint-url http://minio-dev3.minio.io:30948 s3api get-object --bucket awscli-mint-test-bucket-11508 --key awscli-mint-test-bucket-11508-object /tmp/datafile-0-b\n",
  "status": "FAIL",
  "error": "Etag mismatch for multipart 0 byte object"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 17 seconds
(7/15) Running minio-dotnet tests ... done in 28 seconds
(8/15) Running minio-go tests ... done in 36 seconds
(9/15) Running minio-java tests ... done in 38 seconds
(10/15) Running minio-js tests ... done in 45 seconds
(11/15) Running minio-py tests ... done in 1 minutes and 9 seconds
(12/15) Running s3cmd tests ... done in 1 minutes and 0 seconds
(13/15) Running s3select tests ... done in 1 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Review by kannappanr — APPROVED** [review]
LGTM

**Pull request merged**